### PR TITLE
doc: nrf_security: Different configs on secure and non-secure

### PR DESCRIPTION
**Pull request by SebastianBoe**
Documented how to achieve different configuration of nrf_security on
the non-secure and secure side.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

File: nrf_security/doc/different_tfm_config.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      mkdir -p $APP_DIR/secure/zephyr
+      cp $APP_DIR/build/zephyr/include/generated/autoconf.h $APP_DIR/secure/zephyr/
+
+#. Modify the application build scripts such that secure nrf_security
+   instance uses these copied files to configure itself instead of
+   using the same configuration as the non-secure instance.
+
+   .. code-block:: console
+
+      cat ~/ncs/nrf/samples/crypto/nrf_security_snippet.cmake >> $APP_DIR/CMakeLists.txt
```
Comment (tejlmand) [inline nrf_security/doc/different_tfm_config.rst:51]: Did you forget to add `nrf_security_snippet.cmake` to this PR ?

Also I think referring to `~/ncs/` is not optimal.
If we should refer to any default path, then it should probably be the default install path proposed when installing using  Toolchain manager for installing nRF Connect SDK.

File: nrf_security/doc/different_tfm_config.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+#. Copy the resulting configuration from the build directory to the
+   application source directory. This copy will be used to configure
+   the secure nrf_security instance.
+
+   .. code-block:: console
+
+      cp $APP_DIR/build/zephyr/.config $APP_DIR/secure_zephyr.config
+      mkdir -p $APP_DIR/secure/zephyr
+      cp $APP_DIR/build/zephyr/include/generated/autoconf.h $APP_DIR/secure/zephyr/
```
Comment (tejlmand) [inline nrf_security/doc/different_tfm_config.rst:43]: Could we provide a Kconfig option for this, and then use a `add_custom_command(OUTPUT <path>/secure/zephyr/autoconf.h)` without having a `DEPENDS`.

That way `add_custom_command()` will only copy the file if it is missing, but not overwrite an already existing / modified file.

Will save the reader some work, and at the same time allow local modifications.

Having such setting could potentially also remove the need for doing the `cat ... >> CMakeLists.txt` below.

**Review by tejlmand — COMMENTED** [review]
I like the direction.

Clean and simple, and at the same time flexible enough for the user.

But I think we could automate some of the user steps.

File: nrf_security/doc/different_tfm_config.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
+.. _nrf_security_different_tfm_config:
+
+Different configurations of secure and non-secure
+#################################################
+
+Normally in a TF-M application the non-secure app will invoke the PSA
+Crypto API to do crypto. And the feature set of the PSA Crypto API
```
Comment (frkv) [inline nrf_security/doc/different_tfm_config.rst:7]: `PSA Crypto APIs`

File: nrf_security/doc/different_tfm_config.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+1. Build the application for a non-secure board with TF-M.
+
+   .. code-block:: console
+
+      export APP_DIR=~/ncs/nrf/samples/crypto/psa_tls
+      cd $APP_DIR
+      west build -b nrf9160dk_nrf9160_ns
+
+#. Enable nrf_security, if it is not enabled already, and configure
+   the crypto functionality and backends as they should be configured
```
Comment (frkv) [inline nrf_security/doc/different_tfm_config.rst:31]: `the crypto functionality and backends` => `crypto functionality and enabled backends`

Also "the secure side" is probably not a term that is very helpful. Alternatives to this are "SPE", "Secure Partition Environment" or "the secure image". I don't know if we are using a consistent term for this, but we should...

File: nrf_security/doc/different_tfm_config.rst (hunk 1 above)
Comment (tejlmand) [reply nrf_security/doc/different_tfm_config.rst:51]: it's in nrf repo.

File: nrf_security/doc/different_tfm_config.rst (hunk 2 above)
Comment (tejlmand) [reply nrf_security/doc/different_tfm_config.rst:43]: discussed offline.

Can be enhanced later.

**Review by tejlmand — APPROVED** [review]

**Comment by SebastianBoe** [comment]
Waiting for review by @frkv

**Review by frkv — APPROVED** [review]
LGTM

**Comment by SebastianBoe** [comment]
The new nrf_security supports diffferent configs for secure and non-secure so this is no longer needed.

**Pull request closed without merging**